Fix xtreme8000#96 error when starting client with aos version in url

dany-on-demand opened the pull request:
The reason this happens is because network_identifier_split at some point got support for "255.255.255.255" ipv4 format, and if you have the suffix it thinks that it's parsing that. This hotfix works in either case (tested). The information about whether the server is .75 or .76 is not needed as it is already handled heuristically in network_connect and read_PacketWorldUpdate

dany-on-demand committed:
Hotfix xtreme8000#96 error when starting client with aos version in url

xtreme8000 commented:
* There will be out of bounds accesses if `strlen(str)` is less than 5.
* I don't see any need to return `str`

dany-on-demand committed:
make removeSuffix return an error code + check for strlen

dany-on-demand committed:
improve network_identifier_split

remove the "removesuffix" and use less straight forward but more c-like logic. functionality is the same, but it also handles cases where "ip" portion is longer than 32 (like if someone accidentally puts a domain there, which does remind me maybe that should be a feature lol)

dany-on-demand commented:
https://github.com/forked-from-1kasper/tigerspades/commit/247f2ca6ae2870e131ac62135fe3df1e36b082f1 also has a different solution, but like I said as far as I know this is entirely pointless, distinguishing versions at this point is unnecessary and doesn't do anything

xtreme8000 commented on src/network.c:
```diff
@@ -1008,16 +1008,24 @@ int network_identifier_split(char* addr, char* ip_out, int* port_out) {
-		}
-	} else {
+	// parse port, also ignore anything past another ':' (e.g. version info)
+	int next_colon = strchr(port_start + 1, ':');
```
why is next_colon of type `int` even though strchr returns `char*`

xtreme8000 commented on src/network.c:
```diff
@@ -1008,16 +1008,24 @@ int network_identifier_split(char* addr, char* ip_out, int* port_out) {
-	} else {
+	// parse port, also ignore anything past another ':' (e.g. version info)
+	int next_colon = strchr(port_start + 1, ':');
+	*port_out = strtoul(port_start + 1, next_colon, 10);
```
next_colon should be passed as a `char**` pointer, where `strtoul` can write the first invalid char pos to, HOWEVER we don't actually need it here, and can pass `NULL` instead, thus don't even need it in the first place

xtreme8000 replied:
does this PR even do anything different to the original code, except the additional ip length check???
